src/Day04.kt

```diff
@@ -0,0 +1,77 @@
+// https://adventofcode.com/2023/day/4
+import kotlin.math.pow
+
+data class Deck(val currentInstances: Int, val myNumbers: Set<Int>, val benchmark: Set<Int>)
+data class Result(val wonInstances: Int, val deckUpdates: Int)
+
+fun main() {
+    val testInput1 = readInput("Day04_test_1")
+    val testInput2 = readInput("Day04_test_2")
+    val input = readInput("Day04")
+
+    fun part1(input: List<String>) = input
+        .parse1()
+        .map { sets -> sets.first.intersect(sets.second) }
```

Comment:
Ich hätte das bisschen Logik hier auch in Functions auslagern können, aber es ist ja immer noch recht gut lesbar :-)